rust/src/ssh/logger.rs

```diff
@@ -16,49 +16,40 @@
  */
 
 use super::ssh::SSHTransaction;
-use crate::json::*;
+use crate::jsonbuilder::{JsonBuilder, JsonError};
 
-fn log_ssh(tx: &SSHTransaction) -> Option<Json> {
+fn log_ssh(tx: &SSHTransaction) -> Result<Option<JsonBuilder>, JsonError> {
     if tx.cli_hdr.protover.len() == 0 && tx.srv_hdr.protover.len() == 0 {
-        return None;
+        return Ok(None);
     }
-    let js = Json::object();
+    let mut js = JsonBuilder::new_object();
     if tx.cli_hdr.protover.len() > 0 {
-        let cjs = Json::object();
-        cjs.set_string_from_bytes(
-            "proto_version",
-            &tx.cli_hdr.protover,
-        );
+        let mut cjs = JsonBuilder::new_object();
```

Comment:
there is no need to create another object here, you can just `js.open_object("client")?;` and then add the client fields. Then use `js.close()?;` when the client section is done.

The goal is to stream it all into one `JsonBuilder` object so we avoid allocs and copies.

Reply:
ok